### PR TITLE
Added UMD loading for AMD, Node, and globals

### PR DESCRIPTION
**Pull request by deanrad**
The goal of this PR is to allow users of requirejs or other AMD style loaders to depend upon reactive, with dependencies automatically pulled in. Without AMD or Node, it creates globals as before. I plan to add a new example app showing the use of this with requirejs, but wanted to solicit comments on the WIP, including:
- Since it's most consistent with AMD or Node loaders to export _a single_ object, I moved rxt to underneath rx and export only rx
- Npm test will run grunt test _and_ a smoke test that exercises the non-jQuery dependent parts of rx

I'll add the example app shortly...

### CODE REVIEW

**Comment by yang** [comment]
@chicagogrooves on a roll :)

File: src/reactive.coffee [hunk 1]
```diff
@@ -1,865 +1,878 @@
...
+    specialAttrs.class = (elt, value) ->
+      setDynProp elt, 'class', value, (val) ->
+        if _.isString(val) then val else rxt.smushClasses(val)
+
+  rx.rxt = rxt
+  rx
+# end rxFactory definition
+
+do(root = this, factory = rxFactory, deps = ['underscore','underscore.string','jquery']) ->
+  console.log "Invoking factory version 1.03"
```
Comment (yang) [inline src/reactive.coffee]: Stray console.log in the core lib (should try avoiding polluting users' consoles)

**Comment by yang** [comment]
Nice! What's the interim commit about? (required-todos)

**Comment by deanrad** [comment]
Good catch on the console.log. Required-todos is an example application to show the use of reactive with requirejs (it's almost ready). So after a `grunt test`, a `grunt build` should be run to update the files in dist/ before committing is that right ?

**Comment by deanrad** [comment]
Okay - the app is pushed - it'd be helpful if you could pull it down and see if you can run it (otherwise I'll enlist one of my coworkers).. FYI my tip for working with the development version of reactive is to do a `bower link` in the reactive root dir, then do a bower install - and it will use the version in dist instead of the version online.. If it works for you, I'll call this branch done..

**Comment by yang** [comment]
Hey Dean, I appear to be having some trouble merging this back into master - are you able to rebase onto master?

**Comment by deanrad** [comment]
hmm.. checking..

**Comment by deanrad** [comment]
This has been redone in PR #35

**Pull request closed without merging**